source/common/router/config_impl.cc

```diff
@@ -992,10 +992,15 @@ RouteMatcher::RouteMatcher(const envoy::api::v2::RouteConfiguration& route_confi
 RouteConstSharedPtr VirtualHostImpl::getRouteFromEntries(const Http::HeaderMap& headers,
                                                          uint64_t random_value) const {
   // First check for ssl redirect.
-  if (ssl_requirements_ == SslRequirements::ALL && headers.ForwardedProto()->value() != "https") {
+  bool httpRequest = true;
```

Comment:
nit: `http_request`. Also, it seems to me like you are potentially changing the logic here. Should `http_request` default to false if there is no header so we don't do any matching logic? If not, can you add more comments?

Reply:
My intention was to be cautious here and always assume it's an http (i.e not https) unless the header is explicitly set to https.

Reply:
OK that makes sense. Can you add a comment?

Reply:
No prob 👍

Reply:
To elaborate some more, changing it to `if (headers.ForwardedProto() == nullptr) { return nullptr; }` makes sense too, I just do not know all the envoy code paths this code effects, so i chose to do it this way (which seems to me less risky).
if you think that returning null is a better change then I'm happy to change the PR to do that.

Reply:
@alyssawilk thoughts here? (Might need to wait till Monday for those thoughts @yuval-k). That occurred to me also. I think now that you say it I am leaning towards returning nullptr in this case because we assume this header should always be set by the HCM, and if not, there can't be a valid route. Thoughts?

Reply:
Looking at the code, I see that the header is added in `ActiveStream::decodeHeaders` in a `if (!state_.is_internally_created_)` block, which calls `mutateRequestHeaders` (which is the only place i saw that adds the header, but i might have missed something).

I'm not entirely sure what `is_internally_created_`  mean. the comments indicate that `Only sanitize headers on first pass` so I assume its something to do with retries?

It seems more likely that indeed the header is normally set. so i'll change it to return null; worst case i'll change it back if you or @alyssawilk think differently

Reply:
I think at a high level it'd be good to say guaranteed headers are always set (host/authority, forwarded-proto etc) or not always set (always do nullptr checks) and try to be consistent about it.  I'm fine going either way but I'd like to be mindful (this implicitly does the latter) and ideally follow a no-stalebot clean up audit in either direction.